### PR TITLE
refactor(GuildManager): Better handling of creation code

### PR DESCRIPTION
**Pull request by Jiralite**
**Please describe the changes this PR makes and why it should be merged:**
I am sure that this approach is better than setting snake case names and deleting the camel case properties. It's also the approach of other areas of the library and helps tackle #5267.

I noticed `icon` is nullable here, but [Discord's documentation](https://discord.com/developers/docs/resources/guild#create-guild-json-params) doesn't seem to accept a `null` value.

I also reordered the parameters as they are in Discord's documentation.

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

File: packages/discord.js/src/managers/GuildManager.js [hunk 1]
```diff
@@ -164,81 +164,80 @@ class GuildManager extends CachedManager {
...
+        icon: icon && (await DataResolver.resolveImage(icon)),
         verification_level: verificationLevel,
         default_message_notifications: defaultMessageNotifications,
         explicit_content_filter: explicitContentFilter,
-        roles,
-        channels,
+        roles: roles.map(({ color, permissions, ...options }) => ({
+          ...options,
+          color: color && resolveColor(color),
+          permissions: permissions && PermissionsBitField.resolve(permissions).toString(),
```
Comment (kyranet) [inline packages/discord.js/src/managers/GuildManager.js:188]: I'm aware this is the same as before, but permissions won't resolve (and will instead fail) if it's set to the number `0` or the bigint `0n`.

This applies to all the other occurrences (roles, flags...).

File: packages/discord.js/src/managers/GuildManager.js [hunk 2]
```diff
@@ -164,81 +164,80 @@ class GuildManager extends CachedManager {
...
+            rateLimitPerUser,
+            ...options
+          }) => ({
+            ...options,
+            parent_id: parentId,
+            user_limit: userLimit,
+            rtc_region: rtcRegion,
+            video_quality_mode: videoQualityMode,
+            permission_overwrites: permissionOverwrites?.map(({ allow, deny, ...options2 }) => ({
+              ...options2,
```
Comment (kyranet) [inline packages/discord.js/src/managers/GuildManager.js:206]: This doesn't read the outer scope's `options`, you can shadow the variable by renaming `options2` to `options`.

File: packages/discord.js/src/managers/GuildManager.js [hunk 3]
```diff
@@ -164,81 +164,80 @@ class GuildManager extends CachedManager {
...
+        const handleGuild = guild => {
+          if (guild.id === data.id) {
+            clearTimeout(timeout);
+            this.client.removeListener(Events.GuildCreate, handleGuild);
+            this.client.decrementMaxListeners();
+            resolve(guild);
+          }
+        };
+        this.client.incrementMaxListeners();
+        this.client.on(Events.GuildCreate, handleGuild);
```
Comment (kyranet) [inline packages/discord.js/src/managers/GuildManager.js:232]:
```suggestion
        this.client.once(Events.GuildCreate, handleGuild);
```

File: packages/discord.js/src/managers/GuildManager.js [hunk 4]
```diff
@@ -164,81 +164,80 @@ class GuildManager extends CachedManager {
...
     });
 
-    if (this.client.guilds.cache.has(data.id)) return this.client.guilds.cache.get(data.id);
+    return (
+      this.client.guilds.cache.get(data.id) ??
+      new Promise(resolve => {
+        const handleGuild = guild => {
+          if (guild.id === data.id) {
+            clearTimeout(timeout);
+            this.client.removeListener(Events.GuildCreate, handleGuild);
```
Comment (kyranet) [inline packages/discord.js/src/managers/GuildManager.js:226]: With the suggestion below:
```suggestion
```

**Review by kyranet — CHANGES_REQUESTED** [review]

File: packages/discord.js/src/managers/GuildManager.js (hunk 2 above)
Comment (Jiralite) [reply packages/discord.js/src/managers/GuildManager.js]: Couldn't do this due to ESLint no-shadow, so I renamed it to `permissionOverwriteOptions` instead.

File: packages/discord.js/src/managers/GuildManager.js (hunk 1 above)
Comment (Jiralite) [reply packages/discord.js/src/managers/GuildManager.js]: Should be done with 28d9dbaddf173ed3f7317d44faff2ee1fa93bcd8!

**Review by kyranet — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**